crypto/s390x_arch.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef S390X_ARCH_H
+# define S390X_ARCH_H
+
+/*
+ * The elements of OPENSSL_s390xcap_P are the doublewords returned by the STFLE
+ * instruction followed by the doubleword pairs returned by instructions' QUERY
+ * functions. If STFLE returns fewer doublewords or an instruction is not
+ * supported, the corresponding element is zero. The order is as follows:
+ *
+ * STFLE:STFLE:STFLE.
```

Comment:
Shall we throw in additional STFLE? As reservation, or just for even count... On related note, it might make it more readable if we switch to hexadecimal offsets. I mean 0x40 and 0x50 denoting KIMD pair is arguably more readable than 64 and 72. "Readable" in the context means rather "easier to verify".

Reply:
I naturally meant "0x40 and 0x48"...

Reply:
Adding a fourth STFLE is a good idea. Do you refer to the offsets in the asm ?

Reply:
> Do you refer to the offsets in the asm ?

Yes, of course.

Hmmm.... And on second thought one can wonder why not #include this file in asm? Relevant files are already .S, i.e. pre-processed. And if there is one missing, we can convert to .S... This way we wouldn't have to cross-verify the offsets...

Reply:
also thought about this, since having to touch all paces in the asm that reference the capability vector every time for example STFLE returns more bits, is cumbersome and error prone.

Then the extern cap vector variable/struct had to be moved to a seperate header (or repeated in every .c file that needs it).

Reply:
> seperate header

No, doesn't have to, see arm_arch.h for example:
```
#if !__ASSEMBLER__
#endif
```